src/components/Tags/index.tsx

```diff
@@ -145,9 +145,14 @@ const Tags: VoidFunctionComponent<TagsProps> = ({
   const inputRef = useRef<HTMLInputElement>(null)
 
   const dispatchOnChange = (newState: TagsProp) => {
-    const changes = newState.map(tag =>
-      typeof tag === 'object' ? tag?.label : tag,
-    )
+    function isTagDefined(tag: string | null | undefined): tag is string {
```

Comment:
I don't understand why this is necessary ? 🤔

Reply:
Typescript won't infer the type with a normal filter without specifying the `tag is string` part:

![Screenshot 2021-12-23 at 10 14 51](https://user-images.githubusercontent.com/43268759/147217558-88701969-393d-476a-9271-b0f4f1f34056.png)

Reply:
What if you move away from VoidFunctionComponent ?

Reply:
Same issue, if I try to manually type the `filter`, typescript effectively tells me to use a type predicate, so I don't think there's another way to solve this:

![Screenshot 2021-12-23 at 10 23 01](https://user-images.githubusercontent.com/43268759/147218781-0b4f3946-2f54-43bb-9b01-fc4010b59986.png)

Reply:
You also need to set 

```ts
type TagsProp = (
  | string
  | { label: string; index: string }
)[]

```

Reply:
Awesome!